src/paper/signals.py

```diff
@@ -44,36 +46,20 @@ def add_unified_doc(created, instance, **kwargs):
                 print("EXCPETION (add_unified_doc): ", e)
 
 
-@receiver(post_save, sender=Vote, dispatch_uid="recalculate_paper_votes")
+@receiver(post_save, sender=GrmVote, dispatch_uid="recalculate_paper_votes")
 def recalc_paper_votes(sender, instance, created, update_fields, **kwargs):
-    paper = instance.paper
-    new_score = paper.calculate_paper_score()
+    if not isinstance(instance, Paper):
+        return
+    paper = instance
```

Comment:
`paper = instance.paper`

Reply:
Instance will never be paper because the sender of the decorator is `GrmVote`, so instance will always be an instance of `GrmVote`.

```
item = instance.item
if not isinstance(item, Paper):
    return

paper = item
```

Reply:
Instance can be any model isntance that creates a relation with the `sender`. So, it can be Paper

Reply:
https://docs.djangoproject.com/en/4.0/ref/signals/#post-save

Yes, any vote on a paper, hypothesis, or post will call this signal but `instance` will never be the class that created the vote. So to get the correct instance, we need `instance.item`

Reply:
Ah, you are right. 
Also, I just realized that it's safe to remove this signal because the score is being stored at UnifiedDoc & paper no longer has a field named score.